tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -312,3 +313,35 @@ ruleTester.run('no-extraneous-dependencies', rule, {
     }),
   ],
 })
+
+describe('TypeScript', function () {
+  getTSParsers()
+    .filter((parser) => parser !== require.resolve('typescript-eslint-parser'))
+    .forEach((parser) => {
+      const parserConfig = {
+        parser: parser,
+        settings: {
+          'import/parsers': { [parser]: ['.ts'] },
+          'import/resolver': { 'eslint-import-resolver-typescript': true },
+        },
+      }
+
+      ruleTester.run('no-extraneous-dependencies', rule, {
+        valid: [
+          test(Object.assign({
+            code: 'import type { JSONSchema7Type } from "@types/json-schema";',
+            options: [{packageDir: packageDirWithTypescriptDevDependencies, devDependencies: false }],
+          }, parserConfig)),
+        ],
+        invalid: [
+          test(Object.assign({
+            code: 'import { JSONSchema7Type } from "@types/json-schema";',
+            options: [{packageDir: packageDirWithTypescriptDevDependencies, devDependencies: false }],
+            errors: [{
+              message: "'@types/json-schema' should be listed in the project's dependencies, not devDependencies.",
+            }],
+          }, parserConfig)),
+        ],
+      })
+    })
+})
```

Comment:
trailing newline

Reply:
sorry about that one.

Would it be a good idea to add https://eslint.org/docs/rules/eol-last to the eslint config of the project?

Reply:
Yes, it would, thanks; i'll do that.